### PR TITLE
add : amoy flags and genesis

### PR DESCRIPTION
**Pull request by 0xsharma**
# Description

This PR adds the network config/genesis for the new Amoy testnet to the repo. Using this, users will be able to start the node using chain="amoy" command.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,13 +15,13 @@ The easiest way to get started with bor is to install the packages using the com
...
+The network accepts `mainnet`,`amoy` or `mumbai` and the node type accepts `validator` or `sentry` or `archive`. The installation script does the following things:
 - Create a new user named `bor`.
 - Install the bor binary at `/usr/bin/bor`.
 - Dump the suitable config file (based on the network and node type provided) at `/var/lib/bor` and uses it as the home dir.
 - Create a systemd service named `bor` at `/lib/systemd/system/bor.service` which starts bor using the config file as `bor` user.
 
-The releases supports both the networks i.e. Polygon Mainnet and Mumbai (Testnet) unless explicitly specified. Before the stable release for mainnet, pre-releases will be available marked with `beta` tag for deploying on Mumbai (testnet). On sufficient testing, stable release for mainnet will be announced with a forum post.
+The releases supports both the networks i.e. Polygon Mainnet, Amoy and Mumbai (Testnet) unless explicitly specified. Before the stable release for mainnet, pre-releases will be available marked with `beta` tag for deploying on Mumbai/Amoy (testnet). On sufficient testing, stable release for mainnet will be announced with a forum post.
```
Comment (manav2401) [inline README.md:24]: Let's mention Amoy testnet here

File: cmd/geth/config.go [hunk 2]
```diff
@@ -348,6 +352,30 @@ func setDefaultMumbaiGethConfig(ctx *cli.Context, config *gethConfig) {
 	// --pprof is enabled in 'internal/debug/flags.go'
 }
 
+// nolint : wsl
+func setDefaultAmoyGethConfig(ctx *cli.Context, config *gethConfig) {
+	config.Node.P2P.ListenAddr = fmt.Sprintf(":%d", 30303)
+	config.Node.HTTPHost = "0.0.0.0"
```
Comment (manav2401) [inline cmd/geth/config.go:358]: Although this config is not used, 127.0.0.1 is advisable.

File: cmd/geth/config.go [hunk 3]
```diff
@@ -348,6 +352,30 @@ func setDefaultMumbaiGethConfig(ctx *cli.Context, config *gethConfig) {
...
+	config.Node.HTTPVirtualHosts = []string{"*"}
+	config.Node.HTTPCors = []string{"*"}
+	config.Node.HTTPPort = 8545
+	config.Node.IPCPath = utils.MakeDataDir(ctx) + "/bor.ipc"
+	config.Node.HTTPModules = []string{"eth", "net", "web3", "txpool", "bor"}
+	config.Eth.SyncMode = downloader.FullSync
+	config.Eth.NetworkId = 80001
```
Comment (manav2401) [inline cmd/geth/config.go:365]: This needs to be changed.

File: internal/cli/server/chains/amoy.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+			EIP155Block:         big.NewInt(0),
+			EIP158Block:         big.NewInt(0),
+			ByzantiumBlock:      big.NewInt(0),
+			ConstantinopleBlock: big.NewInt(0),
+			PetersburgBlock:     big.NewInt(0),
+			IstanbulBlock:       big.NewInt(0),
+			MuirGlacierBlock:    big.NewInt(0),
+			BerlinBlock:         big.NewInt(0),
+			LondonBlock:         nil,
+			ShanghaiBlock:       nil,
```
Comment (manav2401) [inline internal/cli/server/chains/amoy.go:29]: Shouldn't this be 0 post agra-HF?

**Review by manav2401 — COMMENTED** [review]
LGTM, few minor comments. Nothing affecting the main logic.

File: internal/cli/server/chains/amoy.go (hunk 4 above)
Comment (0xsharma) [reply internal/cli/server/chains/amoy.go:29]: We are going with pre-london deployment and another HF block will enable all the consecutive HFs. That block will be decided and communicated later.

File: cmd/geth/config.go (hunk 2 above)
Comment (0xsharma) [reply cmd/geth/config.go:358]: Duplicated from Mumbai. Maybe this can be taken up in another non-related PR.

File: cmd/geth/config.go (hunk 3 above)
Comment (0xsharma) [reply cmd/geth/config.go:365]: Will take care of it 🤝

File: README.md (hunk 1 above)
Comment (0xsharma) [reply README.md:24]: Sure !

File: internal/cli/server/chains/amoy.go (hunk 4 above)
Comment (manav2401) [reply internal/cli/server/chains/amoy.go:29]: Alright!

File: cmd/geth/config.go (hunk 2 above)
Comment (manav2401) [reply cmd/geth/config.go:358]: Makes sense.

File: README.md (hunk 1 above)
Comment (0xsharma) [reply README.md:24]: We have already mentioned it here.

**Review by anshalshukla — APPROVED** [review]

File: internal/cli/server/chains/amoy.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var amoyTestnet = &Chain{
+	NetworkId: 80002,
+	Genesis: &core.Genesis{
+		Config: &params.ChainConfig{
```
Comment (cffls) [inline internal/cli/server/chains/amoy.go:14]: Can we reuse `AmoyChainConfig` defined in `params/config.go`?

Comment (0xsharma) [reply internal/cli/server/chains/amoy.go:14]: We will use `chains/amoy.go` predominantly as they are compatible with new CLI. `params/config.go` is still being used and for all networks. Will need to define a clear way to deprecate them but in a separate effort.

**Review by cffls — APPROVED** [review]

**Review by temaniarpit27 — APPROVED** [review]

**Comment by 0xsharma** [comment]
Succeeded by #1131 . Closing this PR.

**Pull request closed without merging**